src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1302,12 +1304,17 @@ private IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<Se
             foreach (var group in groupedSelectList)
             {
                 var optGroup = group.First().Group;
-
-                // Wrap if requested.
-                TagBuilder groupBuilder = null;
                 if (optGroup != null)
                 {
-                    groupBuilder = new TagBuilder("optgroup");
+                    var optGroupContent = new BufferedHtmlContent();
```

Comment:
Suggest moving this down closer to the first use (`groupBuilder.InnerHtml = optGroupContent;`)